### ISSUE TITLE
ENOENT for yarn-metadata.json file

### ISSUE THREAD

**Issue by adamreisnz** [yarnpkg/yarn#4321]
Whenever I try to `yarn` I get:

```
error An unexpected error occurred: "ENOENT: no such file or directory, open '/Users/Adam/Library/Caches/Yarn/v1/npm-gulp-4.0.0-alpha.2-311322480cff736e0f423389352d202c97f530e0/.yarn-metadata.json'".
```

I need to remove the `yarn.lock` file for it to work. Then after install, running yarn again fails with the same error.

Not sure why this started happening. This is with yarn 0.28.4 on OS X. I can't upgrade to 1.0.0 due to it failing completely on Heroku.

Please let me know what I can do to help you debug this. The package it's failing on is `"gulp": "git+https://github.com/gulpjs/gulp#4.0",`

**Comment by BYK** [comment]
>I can't upgrade to 1.0.0 due to it failing completely on Heroku.

I'm guessing you are referring to #4320, which we are planning to fix soon.

Does 1.0 work properly on your local?

**Comment by adamreisnz** [comment]
I was afraid to try it yet :)
Had to deal with some production issues and had no time to debug a new major version of yarn.

But I'll give it a try soon and see if that resolves it!

**Comment by adamreisnz** [comment]
Still an issue in 1.0

**Comment by adamreisnz** [comment]
This is still happening in 1.0.2, really annoying, for different projects.

**Comment by bstro** [comment]
I get the same error when trying to deploy to Heroku, but not when running `yarn` locally.

**Comment by ramonakira** [comment]
I have the same issue, using node `lts/boron` and yarn `1.3.2` locally.

**Comment by BYK** [comment]
I wasn't able to reproduce locally so it would be great to have precise reproduction steps here or a small repo that we can try out.

**Comment by luiseduardobrito** [comment]
I got this error today in Yarn v1.6.0, OSX 10.13.4 while linking a dependency installed from a private GitHub repository.

The solution for me was a `yarn cache clean` and a retry, worked like a charm.

**Comment by dbackeus** [comment]
We've been getting this randomly while deploying to Heroku lately. Using node version 6.9.5 with Yarn 1.10.0.

**Comment by SuhairZain** [comment]
Getting the same error on `yarn@0.27.5` and `node@v8.4.0` after I did a `yarn cache clean`. Has anyone figured out a way to fix this?

**Comment by SuhairZain** [comment]
For anyone still struggling with this, my problem was that the shasum checksum for the offending package was wrong in `yarn.lock`. I downloaded the package and calculated the shasum. Changed the checksum in `yarn.lock` to this and it's working now. I remember us having a similar issue before, with this same package and I don't remember what we did to fix it the last time. So this may not be a permanent fix and `yarn` may change to the wrong checksum sometime later, but it works for now.
